template/lib.typ

```diff
@@ -159,40 +167,73 @@
         []
       }
 
-      let level_numbering = if chapter.numbering == none {
+      //section number assignation
+      let item_numbering = if item.numbering == none {
           none
         } else {
-          context numbering(chapter.numbering, ..counter(heading).at(loc))
+          context numbering(item.numbering, ..counter(heading).at(loc))
         }
 
-      let entry = [#grid(columns: (20pt * (chapter.level - 1), 20pt, 1fr, auto), [], level_numbering, body, page_number)]
+      //data representation
 
-      let entry_spacing = if chapter.numbering != none and chapter.level == 1 { 10pt } else { 5pt }
+      //The first 'if' manages the heading configuration (By default it allows you to index the titles of whatever level you want)
+      if(target == heading and (lvl == 1 or item.level <= lvl)){
+        let entry = [#grid(columns: (30pt * (item.level - 1), 30pt + 6pt * (item.level - 1), 1fr, auto), [], item_numbering, body, page_number)]
 
-      show grid: set block(above: entry_spacing, below: entry_spacing)
-
-      if chapter.level == 1 and chapter.numbering != none {
-        strong(entry)
+        let entry_spacing = if item.numbering != none and item.level == 1 { 25pt } else { 5pt }
+      
+        show grid: set block(above: entry_spacing)
+    
+       if (target == heading) or ( item.level <= 2 ){
+          if item.numbering != none {
+
+              if item.level == 1{
+                v(-1em)
+                strong(entry)
+                v(1em) 
+                } else {
+                v(-1em)
+                entry
+                v(1em) 
+               }
+
+          } else {
+             entry
+          }
+       }
+      
+      //The second 'if' helps you with the title levels configuration
+      } else if(target == heading) {[]
+
+      //The last case allows you to manage each type of figure index
       } else {
-        entry
+        let entry_f = [#grid(columns: (4em, 1fr, 1em, 1em), item_numbering,item.caption, h(1em),page_number)]
+
+        entry_f
       }
     }
-}
+  }
```

Comment:
The point of using a selector as `target` is to be able to use any selector generically, that is, making the function compatible with _any_ selector, e.g. for table/figure/heading/abbreviated indexes. Adding specific content to certain selectors makes the behavior of the function more complicated and unpredictable, so we should avoid doing that whenever possible. If you want to introduce behavior that depends on the numbering used on the elements, you could match on whether the element is numerated (has a `numbering` member) instead of the type of the selector used.